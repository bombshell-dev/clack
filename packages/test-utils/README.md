# `@clack/test-utils`

A bunch of useful test utiltiies for use inside clack.
