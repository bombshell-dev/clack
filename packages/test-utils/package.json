{
  "name": "@clack/test-utils",
  "private": true,
  "version": "1.0.0-alpha.1",
  "type": "module",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "types": "./dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bombshell-dev/clack.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/bombshell-dev/clack/issues"
  },
  "homepage": "https://github.com/bombshell-dev/clack/tree/main/packages/test-utils#readme",
  "files": ["dist", "CHANGELOG.md"],
  "author": {
    "name": "James Garbutt",
    "url": "https://github.com/43081j"
  },
  "license": "MIT",
  "packageManager": "pnpm@9.14.2",
  "scripts": {
    "build": "tsc",
    "prepack": "pnpm build"
  },
  "dependencies": {
  },
  "devDependencies": {
  }
}
