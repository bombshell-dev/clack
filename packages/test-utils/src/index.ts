export { MockReadable } from './mock-readable.js';
export { MockWritable } from './mock-writable.js';

export function nextTick(): Promise<void> {
	return new Promise<void>((resolve) => {
		setTimeout(resolve, 0);
	});
}
