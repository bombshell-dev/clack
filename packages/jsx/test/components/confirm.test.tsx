import { MockReadable, MockWritable } from '@clack/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';
import { Confirm } from '../../src/index.js';

describe('Confirm', () => {
	let input: MockReadable;
	let output: MockWritable;

	beforeEach(() => {
		input = new MockReadable();
		output = new MockWritable();
	});

	test('can set message', async () => {
		const element = <Confirm message="foo?" />;
		const task = element.render({ input, output });
		input.emit('keypress', '', { name: 'return' });
		const result = await task;
		expect(result).to.equal(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('can set active text', async () => {
		const element = <Confirm message="foo?" active="DO IT" input={input} output={output} />;
		const task = element.render({ input, output });
		input.emit('keypress', '', { name: 'return' });
		const result = await task;
		expect(result).to.equal(true);
		expect(output.buffer).toMatchSnapshot();
	});

	test('can set inactive text', async () => {
		const element = <Confirm message="foo?" inactive="DONT DO IT" />;
		const task = element.render({ input, output });
		input.emit('keypress', '', { name: 'return' });
		const result = await task;
		expect(result).to.equal(true);
		expect(output.buffer).toMatchSnapshot();
	});
});
