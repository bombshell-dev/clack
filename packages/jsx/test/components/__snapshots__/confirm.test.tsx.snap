// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Confirm > can set active text 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo?
[36m│[39m  [32m●[39m DO IT [2m/[22m [2m○[22m [2mNo[22m
[36m└[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo?
[90m│[39m  [2mDO IT[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`Confirm > can set inactive text 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo?
[36m│[39m  [32m●[39m Yes [2m/[22m [2m○[22m [2mDONT DO IT[22m
[36m└[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo?
[90m│[39m  [2mYes[22m",
  "
",
  "<cursor.show>",
]
`;

exports[`Confirm > can set message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
[36m◆[39m  foo?
[36m│[39m  [32m●[39m Yes [2m/[22m [2m○[22m [2mNo[22m
[36m└[39m
",
  "<cursor.backward count=999><cursor.up count=4>",
  "<cursor.down count=1>",
  "<erase.down>",
  "[32m◇[39m  foo?
[90m│[39m  [2mYes[22m",
  "
",
  "<cursor.show>",
]
`;
